### DISCUSSION TITLE
Convert ttf top woff/woff2

### DISCUSSION THREAD

**Discussion by AndiLeni** [fontforge/fontforge#5128] [Q&A]
Hello,

I have a folder hierarchy with a few hundred folders which contain .ttf files.
What I need is to convert each font file to woff and woff2.

Can I somehow programmatically use fontforge to convert those files from the command line?
I tried the python module, but it seems to be not available.
For accessing fontforge via cli the installation path must be added to the PATH variable I guess?

I am using Windows 10.

Thanks and kind regards

**Comment by skef** [comment]
When you install FontForge on Windows, it doesn't add a python module to the Windows installation (because of ongoing compiler compatibility problems). However, the binary includes a python interpreter. So to run a python script that uses the FontForge module you can run it "through" the FontForge application, on the command line using special command line arguments:

```
fontforge -lang=py -script [scriptname]
```

**Reply by AndiLeni** [reply]
Okay thanks.
The alternative without python would be doing it the way it is described here? https://github.com/fontforge/fontforge/discussions/4754

**Comment by AndiLeni** [comment]
I came up with this solution.
Is there anything which might lead to unexpected behavior?

```ps1
Get-ChildItem *.ttf -Recurse | foreach { 
    $path = $_.DirectoryName+"\"+$_.BaseName+".woff"
    fontforge -lang=ff -c 'Open($1); Generate($2)' $($_.FullName) $path

    $path = $_.DirectoryName+"\"+$_.BaseName+".woff2"
    fontforge -lang=ff -c 'Open($1); Generate($2)' $($_.FullName) $path
}
```

**Reply by skef** [reply]
You asked specifically about Python so my answer focused on that. This is just using FontForge's other built-in scripting language, which is simpler (and less powerful) but also generally fine for this sort of thing. That approach may work fine, or there may be occasional problems if there are "weird" files. You'll have to look at the files to be sure, but there's no reason Python vs "Native" would affect that.

**Reply by AndiLeni** [reply]
No worries, your answer about the python integration was fine :)

Is it possible to ommit all cli output except errors?

**Reply by skef** [reply]
All we have for that is the -quiet flag, which is sort of like that but you don't get to pick and choose.

**Comment by ctrlcctrlv** [comment]
You probably don't want FontForge at all but rather `sfnt2woff` and/or `woff2_compress`. Will save you a TON of problems if any of the fonts have any feature FontForge doesn't support...

**Reply by AndiLeni** [reply]
Thanks, I will have a look at those options.
